source/extensions/clusters/strict_dns/strict_dns_cluster.cc

```diff
@@ -185,7 +190,12 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
 
           if (!response.empty() && parent_.respect_dns_ttl_ &&
               ttl_refresh_rate != std::chrono::seconds(0)) {
-            final_refresh_rate = ttl_refresh_rate;
+            std::chrono::milliseconds jitter =
+                std::chrono::milliseconds(random_generator_.random() % 512); // FIXME
+            if (jitter >= ttl_refresh_rate) {
+              jitter = std::chrono::milliseconds(0);
+            }
+            final_refresh_rate = ttl_refresh_rate - jitter;
```

Comment:
Should we do +- jitter? Or always subtract

Reply:
I did subtract so to avoid having expired entries.